### PR TITLE
Updating README

### LINKED ISSUE #372: GraphQL client not taking available credentials
Hi there,

I was playing with the brand new GraphQL client embedded in the lib but I noticed that it was not taking the credentials. 

Even the Readme suggest that to use the GraphQL step 1-5 are required (getting a token with oath) to work, the doc says otherwise : https://shopify.dev/docs/admin-api/graphql/getting-started 
> Include the access token as a X-Shopify-Access-Token header in all GraphQL queries and mutations

Non working example: 

```python3

import shopify

API_KEY = "mykey"
PASSWORD = "mypassword"
website = "mywebsite"

shopify.ShopifyResource.set_site("https://%s:%s@%s.myshopify.com/admin/api/2020-01" % ( API_KEY, PASSWORD, website))

# returns HTTPError: HTTP Error 401: Unauthorized
shopify.GraphQL().execute("{ shop { name id } }")
```
A simple fix outside of the lib is the following : 
```python3
# manual fix . shopify.ShopifyResource.password is set by the set_url previous call
shopify.ShopifyResource.headers.update({'X-Shopify-Access-Token': shopify.ShopifyResource.password })

# returns expected data
shopify.GraphQL().execute("{ shop { name id } }")
```

This does not happen if we create a shopify session and use it :

```python3
session = shopify.Session("%s.myshopify.com" % website, "2020-01", PASSWORD)
shopify.ShopifyResource.activate_session(session)
```

I believe this should be patched in the lib itself at  is an easy one.
Adding the line : 
```python3
self.headers.update({'X-Shopify-Access-Token': shopify.ShopifyResource.password })
```
after https://github.com/Shopify/shopify_python_api/blob/master/shopify/resources/graphql.py#L10 seems to be enough. 

I can do the PR if you want even if I believe it might take longer to wait for my PR than committing the fix :wink: 

Regards

### PR DESCRIPTION
**Pull request by tanema**
### WHY are these changes introduced?

fixes #372

The readme is 💩  and has always been 💩  and also references a lot of ruby things for some reason. It also does not accurately capture the proper way to authenticate, or validate the authentication.

### WHAT is this pull request doing?
- Updating readme 
  - authentication process
  - graphql access
  - removing unnecessary information
  - making some of it a bit more clear
- Fixing the console setup process so that console running is easier

### CODE REVIEW

**Review by paulomarg — APPROVED** [review]
LGTM, only had a couple of minor suggestions.

File: README.md [hunk 1]
```diff
@@ -273,129 +127,57 @@ product = shopify.Product()
...
 ```
 
-The [tests for this package](https://github.com/Shopify/shopify_python_api/tree/master/test) also serve to provide advanced examples of usage.
-
 ### Prefix options
 
-Some resources such as `Fulfillment` are prefixed by a parent resource in the Shopify API.
-
-e.g. `orders/450789469/fulfillments/255858046`
+Some resources such as `Fulfillment` are prefixed by a parent resource in the Shopify API (e.g. `orders/450789469/fulfillments/255858046`) In order to interact with these resources, you must specify the identifier of the parent resource in your request.
```
Comment (mkevinosullivan) [inline README.md:142]: Missing period before the _In order to ..._
> Some resources such as `Fulfillment` are prefixed by a parent resource in the Shopify API (e.g. `orders/450789469/fulfillments/255858046`). In order to interact with these resources, you must specify the identifier of the parent resource in your request.

File: README.md [hunk 2]
```diff
@@ -273,129 +127,57 @@ product = shopify.Product()
...
-
-e.g. `shopify.Fulfillment.find(255858046, order_id=450789469)`
+```python
+shopify.Fulfillment.find(255858046, order_id=450789469)
+```
 
 ### Console
-
 This package also includes the `shopify_api.py` script to make it easy to
 open up an interactive console to use the API with a shop.
```
Comment (mkevinosullivan) [inline README.md:150]: This package also includes the `shopify_api.py` script to make it easy to
open _~up~_ an interactive console to use the API with a shop.

File: README.md [hunk 3]
```diff
@@ -3,266 +3,120 @@
...
-    There is a method to make the request and get the token for you. Pass
-    all the params received from the previous call (shop, code, timestamp,
-    signature) as a dictionary and the method will verify
-    the params, extract the temp code and then request your token:
+1.  In order to access a shop's data, apps need an access token from that specific shop. We need to authenticate with that shop using OAuth, which we can start in the following way:
 
     ```python
-    token = session.request_token(params)
+    shop_url = "SHOP_NAME.myshopify.com"
+    api_version = '2020-10'
```
[the remark this answers is not in the document]
Comment (tanema) [reply README.md:34]: This is example code and our set example should not be to recommend using the unstable API

**Review by mllemango — APPROVED** [review]
LGTM!

**Review by mkevinosullivan — APPROVED** [review]

**Pull request merged**